apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,370 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.DefinitionTest do
+  alias Lexical.RemoteControl.CodeIntelligence.Definition
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Line
+  alias Lexical.SourceFile.Position
+
+  import Lexical.Test.Fixtures
+  import Lexical.Test.CodeSigil
+  import Line
+
+  use ExUnit.Case, async: false
+
+  setup do
+    start_supervised!(Lexical.SourceFile.Store)
+    Code.put_compiler_option(:ignore_module_conflict, true)
+    :ok
+  end
+
+  defp with_navigation_project(_ctx) do
+    %{project: project(:navigations)}
+  end
+
+  defp with_referenced_file(%{project: project}, file \\ "my_definition.ex") do
+    path = file_path(project, Path.join("lib", file))
+    Code.compile_file(path)
+    %{uri: SourceFile.Path.ensure_uri(path)}
+  end
+
+  defp open_uses_file(project, content) do
+    uri =
+      project
+      |> file_path(Path.join("lib", "my_module.ex"))
+      |> SourceFile.Path.ensure_uri()
+
+    with :ok <- SourceFile.Store.open(uri, content, 1),
+         {:ok, uses_file} <- SourceFile.Store.fetch(uri) do
+      uses_file
+    end
+  end
+
+  defp cursor_to_position(cursor, source_file) do
+    find_line = fn source_file, cursor ->
+      Enum.find(Tuple.to_list(source_file.document.lines), fn line(text: text) ->
+        cursor_full_text = String.replace(cursor, "|", "")
```

Comment:
this is irrelevant, but you're doing this replacement in a loop rather than doing it at the beginning of the loop. this means you're doing `n` allocations rather than 1.
It's also logically incorrect, as you won't be able to work with a line that has both a pipe and a cursor.